src/features/zaaksysteem/service.ts

```diff
@@ -43,32 +44,46 @@ export const useZakenByZaaknummer = (zaaknummer: Ref<string>) => {
   return ServiceResult.fromFetcher(getUrl, overviewFetcher);
 };
 
+const singleZaakFetcher = function fetcher(url: string): Promise<ZaakDetails> {
+  return fetchLoggedIn(url)
+    .then(throwIfNotOk)
+    .then((x) => x.json())
+    .then(mapZaakDetails);
+};
+
 export const useZaakById = (id: Ref<string>) => {
   const getUrl = () => getZaakUrl(id.value);
-
-  function fetcher(url: string): Promise<ZaakDetails> {
-    return fetchLoggedIn(url)
-      .then(throwIfNotOk)
-      .then((x) => x.json())
-      .then(mapZaakDetails);
-  }
-
-  return ServiceResult.fromFetcher(getUrl, fetcher);
+  return ServiceResult.fromFetcher(getUrl, singleZaakFetcher);
 };
 
 export const useZakenByVestigingsnummer = (vestigingsnummer: Ref<string>) => {
   const getUrl = () => {
     if (!vestigingsnummer.value) return "";
     const url = new URL(location.href);
-    url.pathname = zaaksysteemBaseUri + "/zaken";
+    url.pathname = zaaksysteemBaseUri + "/rollen";
     url.searchParams.set(
-      "embedded.rollen.embedded.betrokkeneIdentificatie.vestigingsNummer",
+      "betrokkeneIdentificatie__vestiging__vestigingsNummer",
       vestigingsnummer.value
     );
     return url.toString();
   };
 
-  return ServiceResult.fromFetcher(getUrl, overviewFetcher);
+  const fetcher = (url: string) =>
+    fetchLoggedIn(url)
+      .then(throwIfNotOk)
+      .then(parseJson)
+      .then((r) =>
+        parsePagination(r, async (x) => {
```

Comment:
gaan we rekening houden met meerdere pagina's?

Reply:
Daar ontkomen we denk ik niet aan. Misschien in een losse story vatten? Gaat om alle lijsten uit zaakgericht werken: zaken, klanten, contactmomenten en uiteindelijk contactverzoeken.

Reply:
en daar is ook al een story voor: DIM@5857 - Openzaak paginering
ik heb dit eraan toegevoegd.